### PR TITLE
Add documentation for annotated_function

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Fixes https://github.com/STEllAR-GROUP/hpx/issues/2590.

## Proposed Changes
I have added few lines of documentation for the function. Please let me know if you have any suggestions.

## Any background context you want to provide?
GSoD'19 applicant.

Prev
- https://github.com/STEllAR-GROUP/hpx/pull/3887
- https://github.com/STEllAR-GROUP/hpx/pull/3886
- https://github.com/STEllAR-GROUP/hpx/pull/3885
- https://github.com/STEllAR-GROUP/hpx/pull/3882
- https://github.com/STEllAR-GROUP/hpx/pull/3855
- https://github.com/STEllAR-GROUP/hpx/pull/3854
- https://github.com/STEllAR-GROUP/hpx/pull/3853

### CODE REVIEW

**Comment by hkaiser** [comment]
Please fix the problems reported by inspect (see https://94230-4455628-gh.circle-artifacts.com/0/hpx_inspect_report.html). We have the policy that all our files should have a license and copyright, please add that.

**Comment by msimberg** [comment]
Could you please turn this into doxygen documentation?

**Comment by tapaswenipathak** [comment]
@msimberg I added the doxygen documentation, do you think I can enhance anything else in this PR?

File: hpx/util/annotated_function.hpp [hunk 1]
```diff
@@ -114,7 +114,8 @@ namespace hpx { namespace util
 #endif
 
     ///////////////////////////////////////////////////////////////////////////
-    namespace detail
+    /// \brief Gets useful information for the input function.
```
Comment (msimberg) [inline hpx/util/annotated_function.hpp]: I think we can remove this one. `namespace detail` is implicitly the "internal implementation details"-namespace.

File: hpx/util/annotated_function.hpp [hunk 2]
```diff
@@ -114,7 +114,8 @@ namespace hpx { namespace util
...
+		namespace detail
```
Comment (msimberg) [inline hpx/util/annotated_function.hpp]: Could you please check the indentation in your changes? There are tabs in front of `namespace detail`, while we prefer to use spaces everywhere.

File: hpx/util/annotated_function.hpp [hunk 3]
```diff
@@ -191,7 +206,12 @@ namespace hpx { namespace util
     };
 
     ///////////////////////////////////////////////////////////////////////////
-    template <typename F>
+    /// \brief Given a function as an argument, the user can annotate_function
+    /// as well.
+    /// Annotating includes setting the thread description per thread id.
+    ///
+    /// \param function
+		template <typename F>
```
Comment (msimberg) [inline hpx/util/annotated_function.hpp]: There are also tabs here. Could you please change them to spaces?

**Pull request merged**